azurerm/resource_arm_application_gateway.go

```diff
@@ -547,7 +547,23 @@ func resourceArmApplicationGateway() *schema.Resource {
 					},
 				},
 			},
-
+			"autoscale_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"min_capacity": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"max_capacity": {
+							Type:     schema.TypeInt,
```

Comment:
We should add some validation for greater than zero and some max value. I can't find that information in the portal or the docs but testing for it should be fairly straightforward by just passing in a ridiculously large number. The API normally throws an error and lets the user know what the max value for that attribute is.

Reply:
The portal allows between `min_capacity` and 100 for this,
I've pushed a commit for this which does between 2 (min capacity minimum) and 100.
Unless there's an easy way to get the min capacity value (and its worth it), is this enough?

Reply:
There is a very convoluted way of being able to get that minimum value but this is definitely sufficient. Thanks for taking the time to find this out